.github/workflows/compile_engine.yml

```diff
@@ -0,0 +1,53 @@
+name: Release binaries for circus-engine
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  release:
+    if: startsWith(github.ref, 'refs/tags/')
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          # - windows-latest
+        ocaml-compiler: [4.14.x]
+        
+    runs-on: ${{ matrix.os }}
+    permissions:
+      contents: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+
+      - uses: actions-rs/toolchain@v1
```

Comment:
`actions-rs` is unmaintained. You can use `dtolnay/rust-toolchain` if you want to use a toolchain action.